### PR TITLE
ci: Convert from CircleCI to GH actions

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,8 +17,8 @@ go 1.21
 // github.com/launchdarkly/go-server-sdk/v7 back to "latest" and commit the change.
 
 require (
-	github.com/launchdarkly/go-sdk-common/v3 latest
-	github.com/launchdarkly/go-server-sdk/v7 latest
+	github.com/launchdarkly/go-sdk-common/v3 v3.1.0
```
Comment (louis-launchdarkly) [inline go.mod:20]: Do we still want to keep this `latest` @keelerm84 @cwaldren-ld? This was what "encouraged" people to pick the actual latest version of the SDK.

Comment (keelerm84) [reply go.mod]: Yes, that is my mistake. Restored!

Comment (cwaldren-ld) [reply go.mod]: Now that this is in GHA and terraform, we can switch to Dependabot for updating this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: 1.21
+
+      - name: Build
+        run: |
+          go mod tidy
```
Comment (cwaldren-ld) [inline .github/workflows/ci.yml:32]: There should be no need for `go mod tidy` (if it's complaining that there is, then the go.mod needs to be tidied in a separate PR and then committed)

**Review by cwaldren-ld — APPROVED** [review]
